jigglypuffRL/common/utils.py

```diff
@@ -56,24 +58,67 @@ def evaluate(algo, num_timesteps=1000):
     print("Average Reward: {}".format(total_reward / num_timesteps))
 
 
-def save_params(algo):
-    if algo.save_version is None:
-        torch.save(algo.checkpoint, "{}.pt".format(algo.save_name))
+def save_params(algo, timestep):
+    algo_name = algo.__class__.__name__
+    env_name = algo.env.unwrapped.spec.id
+    directory = algo.save_model
+    path = "{}/{}_{}".format(
+        directory, algo_name, env_name
+    )
+
+    if timestep == algo.save_interval:
+        if not os.path.exists(directory):
+            os.makedirs(directory)
+            log = {}
+            run_num = 0
+        else:
+            f = open("{}/log.pkl".format(directory), "rb")
+            log = pickle.load(f)
+            f.close()
+            run_num = log[path] + 1
+
+        if not os.path.exists(path):
+            os.makedirs(path)
```

Comment:
Rather than the multiple `os.makedirs(path)`, you can do a single `os.makedirs(path, exist_ok=True)` before the if condition

Reply:
The logic is actually different. The first os.path.exists(directory) checks whether there was ever a save made. The second os.path.exists(path) checks if it was made for the specific run of the .py file. They serve different purposes so I can't combine them. If I were to remove it, I would have to do the os.makedirs(path) inside the else too.

Reply:
Got it.